scalrctl/commands/internal/update.py

```diff
@@ -130,14 +151,40 @@ def update():
 
     for index, api_level in enumerate(defaults.API_LEVELS, 1):
 
-        click.echo('[{}/{}] Updating specifications for {} API ... '
+        click.echo('[{}/{}] Updating specifications for Scalr {} API (Swagger)... '
                    .format(index, amount, api_level), nl=False)
 
-        with utils._spinner():
+        with utils.Spinner():
             success, fail_reason = _update_spec(api_level)
 
         if success:
             click.secho('Done', fg='green')
         else:
             click.secho('Failed: {}'.format(fail_reason), fg='red')
 
+
+def update(force=True):
+    """
+    Update CLI.
+    """
+    noupdate_flag_path = os.path.join(defaults.CONFIG_DIRECTORY, ".noupdate")
+    if 'update' in sys.argv or os.path.exists(noupdate_flag_path):
+        if force and os.path.exists(noupdate_flag_path):
+            click.secho('Update is disabled by user. To enable remove %s' % noupdate_flag_path,
+                        fg='yellow')
+
+    if not force:
+        return
```

Comment:
Не очень понятно предназначение опции force т.к. с `force=False` метод просто ничего не делает.

И вобще код апдейта очень мутный, хотелось бы чтобы это выглядело как-то так (не тестил):
```
NOUPDATE_FLAG_PATH = os.path.join(defaults.CONFIG_DIRECTORY, '.noupdate')


def _is_openapi_v2_update_required():
    """
    Determine if OpenAPI v2 update is needed.

    :rtype: bool
    """
    # prevent from running 'update' more than once
    if 'update' in sys.argv or 'configure' in sys.argv or os.path.exists(NOUPDATE_FLAG_PATH):
        return False
    return not all(
        _is_spec_exists(level, 'yaml') and _is_spec_exists(level, 'json')
        for level in defaults.API_LEVELS)


def _is_openapi_v3_update_required():
    """
    Determine if OpenAPI v3 update is needed.

    :rtype: bool
    """
    # prevent from running 'update' more than once
    if 'update' in sys.argv or 'configure' in sys.argv or os.path.exists(NOUPDATE_FLAG_PATH):
        return False
    return not (_is_spec_exists('openapi', 'yaml') and _is_spec_exists('openapi', 'json'))


def _is_openapi_update_required():
    """
    Determine if OpenAPI specification update is needed.

    :rtype: bool
    """
    return _is_openapi_v3_update_required() or _is_openapi_v2_update_required()


def _update_openapi_v2():
    """
    Update OpenAPI v2 specification.

    :return bool: `True` if spec has been updated successfully, otherwise `False`.
    """
    any_failed = False
    for index, api_level in enumerate(defaults.API_LEVELS, 1):
        click.echo((
            "[{}/{}] Updating specifications for Scalr {} API (Swagger)..."
        ).format(index, len(defaults.API_LEVELS), api_level), nl=False)
        with utils.Spinner():
            success, fail_reason = _update_spec(api_level)
        if success:
            click.secho("Done", fg='green')
        else:
            click.secho("Failed: {}".format(fail_reason), fg='red')
            any_failed = True
    return not any_failed


def _update_openapi_v3():
    """
    Update OpenAPI v3 specification.

    :return bool: `True` if spec has been updated successfully, otherwise `False`.
    """
    click.echo("Updating specification for Scalr API (OpenAPI)... ", nl=False)
    with utils.Spinner():
        success, fail_reason = _update_spec("openapi1")
    if success:
        click.secho("Done", fg='green')
        return True
    click.secho("Failed: {}".format(fail_reason), fg='red')
    return False


def update(force=True):
    """
    Update OpenAPI v3 or/and v2 specification (json and yaml).

    :param bool force: Update specification forcefully (default=True).
    """
    if os.path.exists(NOUPDATE_FLAG_PATH):
        if force:
            click.secho((
                "Update is disabled by user. To enable remove {path}"
            ).format(path=NOUPDATE_FLAG_PATH), fg='yellow')
        return
    if not _is_openapi_update_required() and not force:
        return

    if not _update_openapi_v3():
        click.secho("OpenAPI spec is not available.", fg='yellow')
        _update_openapi_v2()
```

Reply:
кул, зарез заімплеменчу